examples/inference/distributed/distributed_speech_generation.py

```diff
@@ -0,0 +1,238 @@
+        texts = [example["text"] for example in examples]
+        original_texts = [example["original_text"] for example in examples]
+
+        return {
+            "input_ids": input_ids,
+            "attention_mask": attention_mask,
+            "ids": ids,
+            "texts": texts,
+            "original_texts": original_texts
+        }
+    def create_dataloader(dataset, batch_size):
+        """Create dataloader with preprocessing"""
+        processed_dataset = [preprocess_fn(item, tokenizer) for item in dataset]
+
+        # Split dataset for distributed processing
+        if distributed_state.num_processes > 1:
+            chunk_size = len(processed_dataset) // distributed_state.num_processes
+            start_idx = distributed_state.process_index * chunk_size
+            end_idx = start_idx + chunk_size if distributed_state.process_index < distributed_state.num_processes - 1 else len(processed_dataset)
+            processed_dataset = processed_dataset[start_idx:end_idx]
+
+        # Create batches
+        batches = []
+        for i in range(0, len(processed_dataset), batch_size):
+            batch = processed_dataset[i:i + batch_size]
+            batches.append(collate_fn(batch))
+        return batches
+
+
+    def save_results(output_queue: queue.Queue, output_dir: pathlib.Path, sampling_rate: int):
+        while True:
+            try:
+                item = output_queue.get(timeout=5)
+                if item is None:
+                    break
+                waveforms, ids, texts, original_texts = item
+                
+                # Save each audio file and its metadata
+                for waveform, file_id, text, original_text in zip(waveforms, ids, texts, original_texts):
+                    # Save audio
+                    wav_path = output_dir / f"{file_id}.wav"
+                    scipy.io.wavfile.write(
+                        wav_path,
+                        rate=sampling_rate,
+                        data=waveform.cpu().float().numpy()
+                    )
+                    
+                    # Save metadata with both truncated and original text
+                    metadata = {
+                        "text_used": text,
+                        "original_text": original_text,
+                        "model": model_name,
+                        "sampling_rate": sampling_rate
+                    }
+                    metadata_path = output_dir / f"{file_id}_metadata.json"
+                    with metadata_path.open("w") as f:
+                        json.dump(metadata, f, indent=4)
+                        
+            except queue.Empty:
+                continue
+
+    # Load and filter data
+    dataset = load_pokemon_data(dataset_split)
+    exist_filter = ExistsFilter(output_dir)
+    dataset = [item for item in dataset if exist_filter(item)]
+    
+    if distributed_state.is_main_process:
+        print(f"Processing {len(dataset)} Pokemon descriptions")
```

Comment:
We can actually just do `distributed_state.print()` instead
```suggestion
    distributed_state.print(f"Processing {len(dataset)} Pokemon descriptions")
```